First pass at adopting CPM

marcpopMSFT opened the pull request:
First pass at adopting CPM. Still need to figure out dependabot and sort the packages.props file.

Then need to see if we can pin items ahead but that'll be a separate PR.

Sending out this PR now to see if we what the test results look like.

marcpopMSFT committed:
First pass at adopting CPM. Still need to figure out dependabot, sort the packages.props file, and figure out the build failure.

Then need to see if we can pin items ahead.

marcpopMSFT committed:
Fix the web tasks build

marcpopMSFT committed:
Merge branch 'main' into marcpopMSFT-transitivepinning

sharwell commented on Directory.Build.props:
```diff
@@ -22,7 +22,7 @@
     <!-- Disable implicit package target fallback, and disable warning for when we explicitly add it (currently needed for
          Microsoft.ApplicationInsights) -->
-    <NoWarn>$(NoWarn);NU1701</NoWarn>
+    <NoWarn>$(NoWarn);NU1701;NU1507</NoWarn>
```
💡 Comment needs to clarify explicitly what NU1701 and NU1507 are. Previously it was implied because only one warning was given.

marcpopMSFT commented on eng/dependabot/Packages.props:
```diff
@@ -4,9 +4,9 @@
-    <PackageReference Update="Verify.DiffPlex" Version="2.2.1" />
-    <PackageReference Update="FakeItEasy" Version="7.4.0" />
-    <PackageReference Update="Wcwidth.Sources" Version="1.0.0" />
+    <PackageVersion Update="Verify.Xunit" Version="20.4.0" />
+    <PackageVersion Update="Verify.DiffPlex" Version="2.2.1" />
+    <PackageVersion Update="FakeItEasy" Version="7.4.0" />
+    <PackageVersion Update="Wcwidth.Sources" Version="1.0.0" />
```
Hadn't had a chance to look into how the msbuild repo did it. Looks like everything is just in the dependabot packages.props folder. Will dependabot handle any props file in that directory regardless of the name? Renaming and importing works for me.

marcpopMSFT replied:
Left it as packages.props but imported it in the up-level directory.packages.props file.

marcpopMSFT commented on eng/Analyzers.props:
```diff
@@ -1,5 +1,5 @@
 <Project>
   <ItemGroup Condition="'$(DotNetBuildFromSource)' != 'true'">
-    <PackageReference Include="StyleCop.Analyzers" Version="$(StyleCopAnalyzersPackageVersion)" PrivateAssets="all" Condition="'$(EnableStyleCopAnalyzer)' == 'true'" />
+    <PackageVersion Include="StyleCop.Analyzers" Version="$(StyleCopAnalyzersPackageVersion)" PrivateAssets="all" Condition="'$(EnableStyleCopAnalyzer)' == 'true'" />
```
There aren't actually use sites. The props is imported in the D.B.props and four projects set EnableStyleCopAnalyzer to true. I'll have to switch this back to a packagereferene and then add a package version in the main packages props.

marcpopMSFT commented on src/Tests/Microsoft.NET.TestFramework/TestCommandLine.cs:
```diff
@@ -9,9 +9,9 @@ public class TestCommandLine
 
-        public bool UseFullFrameworkMSBuild { get; private set; }
+        public bool UseFullFrameworkMSBuild { get; private set; } = true;
 
-        public string FullFrameworkMSBuildPath { get; private set; }
+        public string FullFrameworkMSBuildPath { get; private set; } 
```
that's what I get for not cleaning up my branch after a prior investigation

marcpopMSFT committed:
Respond to PR feedback:
- Added comment for NoWarn
- Change how Packages.props is imported
- Change how analyzers are imported
- fix sdks build
- remove fullframework true

marcpopMSFT committed:
Merge branch 'main' into marcpopMSFT-transitivepinning

MiYanni commented on src/Tests/Microsoft.NET.TestFramework/TestCommandLine.cs:
```diff
@@ -11,7 +11,7 @@ public class TestCommandLine
         public bool UseFullFrameworkMSBuild { get; private set; }
 
-        public string FullFrameworkMSBuildPath { get; private set; }
+        public string FullFrameworkMSBuildPath { get; private set; } 
```
nit: space

marcpopMSFT committed:
Remove extra space
